tests/test_create_downloader.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+
+from datasetinsights.io.downloader.base import create_downloader
+from datasetinsights.io.downloader.http_downloader import HTTPDownloader
+from datasetinsights.io.downloader.unity_simulation import (
+    UnitySimulationDownloader,
+)
+
+
+@pytest.mark.parametrize(
+    "source_uri", ["http://", "https://"],
+)
+def test_create_downloader_http_downloader(source_uri):
+    # arrange
+
+    # act
+    downloader = create_downloader(source_uri=source_uri)
+
+    # assert
+    assert isinstance(downloader, HTTPDownloader)
+
+
+def test_create_downloader_unity_simulation_downloader():
```

Comment:
Why test create_downloader when `_find_downloader` is tested against different cases? What about cases when _find_downloader can find a downloader for a given source_uri? What about cases when source_uri is none?

Reply:
Good point, there is no test for create_downloader with input None. I don't understand what you mean by the two first questions though.

Reply:
I don't think it makes sense to unit test _find_downloader as it is a private function

Reply:
We should have unittest for ANY function that preform non-trivial operations, whether it's public or private function. The logic inside `create_downloader` seems trivial however `_find_downloader` is non trivial.  

Typo in my previous comment. Correction: 

> What about cases when `_find_downloader` can't find a `DatasetDownloader` from a given source_uri? What about cases when source_uri is none?

Reply:
Added these two tests that test the interface.


```
def test_create_downloader_invalid_input():
    # arrange
    source_uri = "invalid_protocol://"
    # assert
    with pytest.raises(ValueError):
        # act
        create_downloader(source_uri=source_uri)


def test_create_downloader_none_input():
    # arrange
    source_uri = None
    # assert
    with pytest.raises(TypeError):
        # act
        create_downloader(source_uri=source_uri)
```

Reply:
I think we should focus on testing the interface instead of the implementation

Reply:
What kind of cases is missing?